flow/include/flow/Traceable.h

```diff
@@ -177,6 +179,11 @@ template <class T>
 struct TraceableStringImpl : std::true_type {
 	static constexpr bool isPrintable(char c) { return 32 <= c && c <= 126; }
 
+	template <class Str>
+	static std::string toString(std::filesystem::path& value) {
+		return value.string();
+	}
+
```

Comment:
I see you are trying to "overload" the `toString` method. However, contrary to intuition, C++ function templates do not overload. So there is no reason you add the `template` here. Herb Sutter has a very comprehensive explanation on this topic: http://www.gotw.ca/gotw/049.htm. Plus you are trying to capture `value` using a non-const reference. This causes all r-value calls fail, e.g. `toString(std::filesystem::path())` will not compile. Usually a better choice is to use `const std::filesystem::path&` if `std::filesystem::path::string()` is const-constrainted. The other choice is to use rvalue-reference, which could be more flexible, yet more subtle and difficult to understand (sorry I do not know your level of knowledge of C++, if you understand rvalue-reference please ignore what i said.) In this particular situation, I would suggest you implement the `template struct TraceableString<std::system::path>` rather than this "overload". Another choice is to use
```c++
if constexpr (std::is_same<Str, std::filesystem::path>::value) {
  return value.string()
} else {
 // old code
}
```
but of course this is not recommended.

Reply:
Thank you so much for providing the in-depth explanation... This is super helpful! Let me add this change here and revert suggestion above

Reply:
Also, had a few concerns; I think that this issue may be beyond my capabilities. I'd like to still work on it but this would take a considerably a lot more time for me. At this point I'm considering to go back on previous commits to test out smaller changes instead of what is here since it's a bit too much to trace what's breaking what.

Curious to hear your thoughts on this :)

Reply:
I help if it is needed but at the end of week, what are your concern? Avoid complexity.

Reply:
> I help if it is needed but at the end of week, what are your concern? Avoid complexity.


Remove this function and implement template specialization `TraceableString<std::system::path>`